aeneas/src/core/Eval.v3

```diff
@@ -844,13 +844,23 @@ def evalOp(op: Operator, args: Arguments) -> Result {
 			var ref = args.ref(0);
 			var index = args.i(1);
 			if (u32.view(index) >= u32.view(max)) return args.throw(V3Exception.BoundsCheck, null);
-			return doRefLayoutGetField(args, ref, offset + scale * index);
+			return doRefLayoutGetField(args, args.getTypeArg(1), ref, offset + scale * index);
 		}
 		RefLayoutSetRepeatedField(offset, scale, max) => {
 			var ref = args.ref(0);
 			var index = args.i(1);
 			if (u32.view(index) >= u32.view(max)) return args.throw(V3Exception.BoundsCheck, null);
-			return doRefLayoutSetField(args, ref, offset + scale * index, args.getArg(2));
+			return doRefLayoutSetField(args, args.getTypeArg(1), ref, offset + scale * index, args.getArg(2));
+		}
+		ByteArrayGetField(offset) => {
+			var array = args.r(0);
+			var i_offset = args.i(1);
+			return doRefLayoutGetField(args, args.getTypeArg(0), ByteArrayOffset.new(array, offset), i_offset);
```

Comment:
Can you refactor the underlying doRefLayoutGetField to take the array and offset individually to avoid creating an intermediary `ByteArrayOffset` object?

Reply:
The underlying `doRefLayoutGetField` is taking an offset since the (array, offset) pair is being passed around as a single value when evaluating without normalization. There are calls to methods on `ByteArrayOffset` in `doRefLayoutGetField`, so we'll have to lift those methods out of the class to avoid creating an object. Should we still do it?

Reply:
Ok, yeah, I see that now. It's OK to keep as is and drop in an `XXX`.